### DISCUSSION TITLE
backtest open more position

### DISCUSSION THREAD

**Discussion by dalet78** [kernc/backtesting.py#681] [Q&A]
Hi,
in the first I want to say you, thanks for this library.
I have an error that I encountered, 
I receive form my strategy a dataframe with signal ( 1 for buy position and 2 for sell position)
I transform my dataframe in array. I learned your library and you use array for to compute backtest.
I verified my array and I have only 16 signal, however when I run the backtest i found 1504 trades? 
Please can you help me to understand my error?
![image](https://user-images.githubusercontent.com/45757581/178341146-fe8a93d7-15ab-45e9-98da-2a0cd6b356c0.png)

this is my code:
```

from backtesting import Strategy, Backtest
import pandas as pd


class MyStrategy(Strategy):

    def init(self):
        super().init()
        # print (type(SIGNAL(self.data.df)))
        self.signal = pd.Series(SIGNAL(self.data.df)).array


    def next(self):
        # super().next()
        if self.signal==1:
            sl1 = self.data.Low[-1] - 200e-4
            tp1 = self.data.Close[-1] + 400e-4
            self.buy(sl=sl1, tp=tp1)
        elif self.signal==2:
            sl1 = self.data.Low[-1] + 2000e-4
            tp1 = self.data.Close[-1] - 4000e-4
            self.sell(sl=sl1, tp=tp1)

def backtest(df):
    bt = Backtest(df, MyStrategy, cash=10_000, commission=.002, exclusive_orders=True)
    stat = bt.run()
    print(stat)
    bt.plot()

def SIGNAL(df):
    return df.signal 

```

**Comment by kernc** [comment]
Firstly, you have an issue in line:
```py
        self.signal = pd.Series(SIGNAL(self.data.df)).array
```
If you want `self.signal` to be auto-extended to correct length at each iteration of `next()` (and conditions such as `if self.signal == 1` to evaluate correctly), the line should instead read:
```py
        self.signal = self.I(SIGNAL, self.data.df)
```
See [`Strategy.I()`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.I) wrapper API reference or the relevant section of the [Backtesting.py Quick Start tutorial](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Strategy).

**Reply by dalet78** [reply]
thanks works perfectly now
